ush/python/pygfs/utils/marine_da_utils.py

```diff
@@ -74,3 +74,95 @@ def stage_ens_mem(task_config: AttrDict) -> None:
     letkf_stage_list = parse_j2yaml(task_config.MARINE_ENSDA_STAGE_BKG_YAML_TMPL, ensbkgconf)
     logger.info(f"{letkf_stage_list}")
     FileHandler(letkf_stage_list).sync()
+
+
+@logit(logger)
+def test_hist_date(histfile: str, ref_date: datetime) -> None:
+    """
+    Check that the date in the MOM6 history file is the expected one for the cycle.
+    TODO: Implement the same for seaice
+    """
+
+    ncf = Dataset(histfile, 'r')
+    hist_date = dparser.parse(ncf.variables['time'].units, fuzzy=True) + timedelta(hours=int(ncf.variables['time'][0]))
+    ncf.close()
+    logger.info(f"*** history file date: {hist_date} expected date: {ref_date}")
+    assert hist_date == ref_date, 'Inconsistent bkg date'
```

Comment:
`raise ValueError` if there is a mismatch.  `assert` suggested use is in tests and should not be used in production code.  
Use `FATAL ERROR` language when raising errors.